app/javascript/src/Users/utils/index.jsx

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import type { Feature } from 'Users/types'
+
+const FEATURE_NAMES: { [feature: Feature]: string } = {
+  portal: 'Developer Portal',
+  finance: 'Billing',
+  settings: 'Settings',
+  partners: 'Developer Accounts -- Applications',
+  monitoring: 'Analytics',
+  plans: 'Integration & Application Plans',
+  policy_registry: 'Policy Registry'
+}
+
+const FEATURES_GRANTING_SERVICE_ACCESS = ['partners', 'monitoring', 'plans']
+
+export function getFeatureName (feature: Feature): string {
+  if (feature in FEATURE_NAMES) {
+    return FEATURE_NAMES[feature]
+  }
+
+  throw new Error(`${feature} is not a known feature`)
+}
+
+export function isServicePermissionsGranted (features: Feature | 'services' | Array<Feature | 'services'>): boolean {
+  if (typeof features === 'string') {
+    return FEATURES_GRANTING_SERVICE_ACCESS.includes(features)
+  } else {
+    return !!features.find(isServicePermissionsGranted)
+  }
+}
+
+export function toggleElementInCollection<T> (el: T, collection: T[]): T[] {
+  const i = collection.indexOf(el)
+
+  if (i > -1) {
+    collection.splice(i, 1)
```

Comment:
This will be used to set a new state, right? Maybe we could avoid modifying the collection and returning a new Array. I feel uneasy when I see `splice` and `push` methods hehe